src/toolkits/object_detection/object_detector.cpp

```diff
@@ -539,6 +554,132 @@ variant_map_type object_detector::perform_evaluation(gl_sframe data,
+                               /* stride_width */        1,
+                               /* padding */             padding_type::SAME,
+                               /* weight_init_fn */      xavier_weight_initializer(x.second*3*3, x.second*3*3, &random_engine));
+
+      // Append batchnorm7.
+      nn_spec->add_batchnorm(/* name */                  "batchnorm"+std::to_string(x.first)+"_fwd",
+                             /* input */                 "conv"+std::to_string(x.first)+"_fwd",
+                             /* num_channels */          x.second,
+                             /* epsilon */               0.00001f);
+
+      // Append leakyrelu7.
+      nn_spec->add_leakyrelu(/* name */                  "leakyrelu"+std::to_string(x.first)+"_fwd",
+                             /* input */                 "batchnorm"+std::to_string(x.first)+"_fwd",
+                             /* alpha */                 0.1f);
+
+      num_features = x.second;
+  }
+
+  // Append conv7, initialized using the Xavier method (with base magnitude 3).
+  // The conv7 weights have shape [1024, 1024, 3, 3], so fan in and fan out are
+  // both 1024*3*3.
+  xavier_weight_initializer conv7_init_fn(1024*3*3, 1024*3*3, &random_engine);
+  nn_spec->add_convolution(/* name */                "conv7_fwd",
+                           /* input */               "leakyrelu6_fwd",
+                           /* num_output_channels */ 1024,
+                           /* num_kernel_channels */ 1024,
+                           /* kernel_height */       3,
+                           /* kernel_width */        3,
+                           /* stride_height */       1,
+                           /* stride_width */        1,
+                           /* padding */             padding_type::SAME,
+                           /* weight_init_fn */      conv7_init_fn);
+
+  // Append batchnorm7.
+  nn_spec->add_batchnorm(/* name */                  "batchnorm7_fwd",
+                         /* input */                 "conv7_fwd",
+                         /* num_channels */          1024,
+                         /* epsilon */               0.00001f);
+
+  // Append leakyrelu7.
+  nn_spec->add_leakyrelu(/* name */                  "leakyrelu7_fwd",
+                         /* input */                 "batchnorm7_fwd",
+                         /* alpha */                 0.1f);
+
+  // Append conv8.
+  static constexpr float CONV8_MAGNITUDE = 0.00005f;
+  const size_t num_classes = read_state<size_t>("num_classes");
+  const size_t num_predictions = 5 + num_classes;  // Per anchor box
+  const size_t conv8_c_out = anchor_boxes().size() * num_predictions;
+  auto conv8_weight_init_fn = [&random_engine](float* w, float* w_end) {
+    std::uniform_real_distribution<float> dist(-CONV8_MAGNITUDE,
+                                               CONV8_MAGNITUDE);
+    while (w != w_end) {
+      *w++ = dist(random_engine);
```

Comment:
I'm a big believer that if you're ever relying on the difference between pre and post incrementing, you're trying to do too much on one line.